app/controllers/api/v1/allocation_events_controller.rb

```diff
@@ -3,15 +3,27 @@
 module Api
   module V1
     class AllocationEventsController < ApiController
-      before_action :validate_params, only: :create
-      after_action :send_move_notifications, only: :create
+      include Eventable
 
-      def create
+      def cancel
+        validate_params!(cancel_params)
+
+        allocation.transaction do
+          allocation.cancel(cancellation_details)
+          process_event(allocation.moves, Event::CANCEL, cancel_move_params)
+        end
+
+        render status: :no_content
+      end
+
+      def events
+        # TODO: this method should be deleted, but kept here until the front end is updated
+        validate_params!(event_params)
         case event_name
         when 'cancel'
           allocation.cancel
+          send_move_notifications
```

Comment:
any reason why this can't go via the event log, the same as the `cancel` method above ?
```
        allocation.transaction do
          allocation.cancel(cancellation_details)
          process_event(allocation.moves, Event::CANCEL, cancel_move_params)
        end
```
? I know it will be deleted soon, but would be nice if it did the same thing under the hood.

Reply:
I'd prefer not to mess with this existing method - it's not creating events at present and is used in production so is safer to switch to the new API endpoint and then start using the events approach.

Reply:
👍